sql/core/src/main/scala/org/apache/spark/sql/sources/interfaces.scala

```diff
@@ -458,7 +458,12 @@ abstract class HadoopFsRelation private[sql](
 
   private var _partitionSpec: PartitionSpec = _
 
-  private[sql] def bucketSpec: Option[BucketSpec] = None
+  private[this] var malformedBucketFile = false
```

Comment:
Comment. Can we make a test case for this? Maybe an insert into non-bucketed table into the same directory.